lib/features/email/presentation/widgets/pdf_viewer/pdf_viewer.dart

```diff
@@ -137,36 +137,84 @@ class _PDFViewerState extends State<PDFViewer> {
                   valueListenable: _pdfViewStateNotifier,
                   builder: (context, viewState, widget) {
                     if (viewState is DownloadAttachmentForWebSuccess) {
-                      return PdfViewer.openData(
+                      return PdfViewer.data(
                         viewState.bytes,
-                        viewerController: _pdfViewerController,
-                        onError: (error) {
-                          logError('_PDFViewerState::build:openData:onError:: $error');
-                          _pdfViewStateNotifier.value = DownloadAttachmentForWebFailure(exception: error);
-                        },
+                        sourceName: this.widget.attachment.generateFileName(),
+                        controller: _pdfViewerController,
                         params: PdfViewerParams(
                           panAxis: PanAxis.vertical,
                           scrollByMouseWheel: 0.5,
-                          layoutPages: (viewSize, pages) {
-                            List<Rect> rect = [];
-                            final viewWidth = viewSize.width;
-                            final viewHeight = viewSize.height;
-                            final maxHeight = pages.fold<double>(0.0, (maxHeight, page) => max(maxHeight, page.height));
-                            final maxWidth = pages.fold<double>(0.0, (maxWidth, page) => max(maxWidth, page.width));
-                            final ratio = viewHeight / max(maxHeight, maxWidth);
-                            log('_PDFViewerState::build: viewWidth = $viewWidth | viewHeight = $viewHeight | maxHeight = $maxHeight | ratio = $ratio');
-                            var top = 0.0;
-                            double padding = 16.0;
-                            for (var page in pages) {
-                              final width = page.width * ratio;
-                              final height = page.height * ratio;
-                              final left = viewWidth > viewHeight ? (viewWidth / 2) - (width / 2) : 0.0;
-                              rect.add(Rect.fromLTWH(left, top, width, height));
-                              top += height + padding;
+                          backgroundColor: Colors.transparent,
+                          onViewerReady: (_, __) {
+                            _pdfViewerController.setZoom(
+                              _pdfViewerController.centerPosition,
+                              1);
+                            setState(() => _showPagination = true);
+                          },
+                          minScale: 1,
+                          maxScale: 4,
+                          boundaryMargin: const EdgeInsets.all(double.infinity),
```

Comment:
It seems the margin is not correct. Can you explain it.

Reply:
Without it pdf viewer cannot zoom out to the minimum scale

Reply:
So we should set minimum scale value to zoom out. Currently, PDF Viewer is shrinking too much.

<img width="1438" alt="Screenshot 2024-12-16 at 13 46 22" src="https://github.com/user-attachments/assets/0a9a4d64-a1b6-4eaa-b0d5-920bafb1e807" />

Reply:
https://github.com/user-attachments/assets/131ba10c-ac83-4fba-8d96-61aad268590b